extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -63,6 +65,16 @@ public abstract class IcebergBaseLayout implements TableLocationKeyFinder<Iceber
      */
     ParquetInstructions parquetInstructions;
 
+    /**
+     * The number of files discovered, used for ordering the keys.
+     */
+    private int fileCount;
+
+    /**
+     * Create a new {@link IcebergTableLocationKey} for the given file URI.
+     * <p>
+     * This method assumes that keys are created in a specific order and that each key is created only once.
```

Comment:
This works because our locationKey creation happens from inside a synchronized method through a cache.